core/deployment/src/main/java/io/quarkus/deployment/ExtensionLoader.java

```diff
@@ -620,7 +621,8 @@ private static Consumer<BuildChainBuilder> loadStepsFromClass(Class<?> clazz,
                                 .andThen(bsb -> bsb.consumes(buildItemClass, ConsumeFlags.of(ConsumeFlag.OPTIONAL)));
                         methodParamFns.add((bc, bri) -> (Supplier<Optional<? extends SimpleBuildItem>>) () -> Optional
                                 .ofNullable(bc.consume(buildItemClass)));
-                    } else if (rawTypeOf(parameterType) == Executor.class) {
+                    } else if (rawTypeOf(parameterType) == Executor.class
+                            || rawTypeOf(parameterType) == ExecutorService.class) {
```

Comment:
This seems risky to me since `bc.getExecutor()` returns an `Executor` instead of an `ExecutorService`. Why not change it as well?

Reply:
It's already changed. Do you mean [this one](https://github.com/quarkusio/quarkus/pull/27159/files#diff-0d5e000a85200fc232c5a6bbbb4e1e4a82f6f7b04901e09fc9957e08d9b40f63R224-R229)?

Reply:
Ah missed it, sorry

Reply:
np :-)